BoxContentSDK/BoxContentSDK/BOXNSURLSessionManager.h

```diff
@@ -10,6 +10,57 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@protocol BOXNSURLSessionTaskDelegate <NSObject>
+
+/**
+ * To be called to finish the operation for a NSURLSessionTask upon its completion
+ *
+ * @param response  The response received from Box as a result of the API call.
+ * @param error     An error in the NSURLErrorDomain
+ */
+- (void)finishURLSessionTaskWithResponse:(NSURLResponse *)response error:(NSError *)error;
+
+@optional
+
+/**
+ * To be called to process the intermediate response from the task
+ *
+ * @param response  The intermediate response received from Box as a result of the API call
+ */
+- (void)processIntermediateResponse:(NSURLResponse *)response;
+
+/**
+ * To be called to process the intermediate data from the task
+ *
+ * @param data  The intermediate data received from Box as a result of the API call
+ */
+- (void)processIntermediateData:(NSData *)data;
+
+@end
+
+@protocol BOXNSURLSessionDownloadTaskDelegate <BOXNSURLSessionTaskDelegate>
+
+@optional
+
+/**
+ * Output stream to write data into. If destinationPath is provided, outputStream will be ignored
+ * Using outputStream to consume data will not allow the request to be executed in the background if the app is killed/suspended
+ */
+- (NSOutputStream *)outputStream;
+
+/**
+ * The location for output file. If provided, outputStream will be ignored
+ * Using destinationPath to consume data will allow request to be executed in the background if the app is killed/suspended and resume upon app restarts/resumes
+ */
+- (NSString *)destinationPath;
+
+/**
+ * To be called to report ongoing progress of the task
+ */
+- (void)progressWithExpectedTotalBytes:(long long)expectedTotalBytes totalBytesReceived:(unsigned long long)totalBytesReceived;
```

Comment:
The order of received before expected makes more sense to me, like `progressWithTotalBytesRecieved:expected:`. Why is one of these signed, and one unsigned? I'm guessing you get a negative expected if we don't know? Should we be using `int64_t` to be compatible with NSURLSession?